fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/client/item/v1/UpdateAnimationHandler.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.item.v1;
+
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Provides information on whether a {@link net.minecraft.item.Item} should run its NBT update animation.
+ */
+@Environment(EnvType.CLIENT)
+public interface UpdateAnimationHandler {
+	/**
+	 * Called once per tick to check whether the given held {@link ItemStack} should run the update animation.
+	 *
+	 * <p>In vanilla, the update animation is run if an {@link ItemStack} has any changed NBT values.
+	 * The second {@link ItemStack} parameter is the most up-to-date version, while the original {@link ItemStack} is a cached version from the previous tick.
+	 * Both {@link ItemStack} instances are guaranteed to have the same underlying {@link net.minecraft.item.Item}.
+	 *
+	 * @param original  cached {@link ItemStack} from previous tick
+	 * @param updated  updated {@link ItemStack} to check for updates on
+	 * @return  whether or not the item update animation should play
+	 */
+	@Environment(EnvType.CLIENT)
+	boolean updateAnimation(ItemStack original, ItemStack updated);
+}
```

Comment:
Some suggestions:
* First sentence should clearly explain what this does: `Return whether the item stack update animation should be played this tick.` and then detail more.
* Javadoc spacing is weird, it should only be 1 space after `@return` and `@param`.
* Rename to `allowUpdateAnimation` to make what this does 100% explicit.
* Instead of `return whether...`, I would say `return true if ..., and false otherwise.` for more clarity.

Reply:
`Item stack update animation` is probably too vague and may need to be more clearly explained. I had to run the testmod to understand what this was about.

Reply:
Imo the class name is fine. `AnimationUpdateHandler` is slightly better, but we do have names like `UpdateTagsS2CPacket` in yarn.